lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -903,7 +903,7 @@ def post_data(action, parameters = {})
       end
 
       def error_code_from(response)
-        STANDARD_ERROR_CODE_MAPPING[response['errorCode']] || response['errorCode']
+        response.dig('additionalData', 'refusalReasonRaw') || response['refusalReason'] || STANDARD_ERROR_CODE_MAPPING[response['errorCode']] || response['errorCode']
```

Comment:
Yeah in a separate PR with ticket ECS-3421 I removed the digging for the different fields from core. We use the function `error_code_from(am_response)` on every core gateway though so we can't remove the whole function.

Reply:
I think response['refusalReason'] should be moved to the end. This is because response['refusalReason']  will only happen if everything else is null it won't update anything that is already occuring.